---
title: "Haplotype networks" 
output: html_notebook
---

Illingworth, C., Hamilton, W. L., Jackson, C., Warne, B., Popay, A., Meredith, L., Hosmillo, M., Jahun, A., Fieldman, T., Routledge, M., Houldcroft, C. J., Caller, L., Caddy, S., Yakovleva, A., Hall, G., Khokhar, F. A., Feltwell, T., Pinckert, M. L., Georgana, I., Chaudhry, Y., … Török, M. E. (2022). A2B-COVID: A Tool for Rapidly Evaluating Potential SARS-CoV-2 Transmission Events. Molecular biology and evolution, 39(3), msac025. https://doi.org/10.1093/molbev/msac025


## Get the sequence data from COG-UK

```{bash include=T, eval=T, echo=T, warning=T, message=T}

if [ -s data_from_cog-uk/EXET_seqs_01_10_20_to_28_02_21.fna ]; then
  echo data_from_cog-uk/EXET_seqs_01_10_20_to_28_02_21.fna already exists
  
elif [ -s data_from_cog-uk/cog_alignment.fasta ]; then
  echo data_from_cog-uk/cog_alignment.fasta already exists
  python3 get_sequences_within_date_range.py > data_from_cog-uk/EXET_seqs_01_10_20_to_28_02_21.fna
  
else
  wget https://cog-uk.s3.climb.ac.uk/phylogenetics/latest/cog_alignment.fasta.gz
  gunzip cog_alignment.fasta.gz
  mv cog_alignment.fasta data_from_cog-uk/
  python3 get_sequences_within_date_range.py > data_from_cog-uk/EXET_seqs_01_10_20_to_28_02_21.fna
  fi
  
```

## Get metadata from COG-UK

```{bash include=T, eval=T, echo=T, warning=T, message=T}
if [ -s cog_metadata.csv ]; then
  echo cog_metadata.csv already exists
elif [ -s cog_metadata.csv.gz ]; then
  echo cog_metadata.csv.gz already exists
  gunzip cog_metadata.csv.gz
else
  wget https://cog-uk.s3.climb.ac.uk/phylogenetics/latest/cog_metadata.csv.gz
  gunzip cog_metadata.csv.gz
  fi
```

## Generate outbreak individual outbreak files from the combined file

```{bash include=T, eval=T, echo=T, warning=T, message=T}
rm outbreak.*.tsv
ln -s outbreak_lists/patientstaff_data_for_haplotype_network.txt .
perl generate_individual_outbreaks.pl patientstaff_data_for_haplotype_network.txt
mv outbreak.*.tsv outbreak_lists/
rm patientstaff_data_for_haplotype_network.txt
```

### Generate nexus files for each outbreak

```{bash include=T, eval=T, echo=T, warning=T, message=T}

ln -s outbreak_lists/*.tsv .
ln -s outbreak_lists/patientstaff_data_for_haplotype_network.txt .
ln -s data_from_cog-uk/EXET_seqs_01_10_20_to_28_02_21.fna .

for outbreak_file in outbreak.*.tsv patientstaff_data_for_haplotype_network.txt
  do echo $outbreak_file
  perl get_haplotypes_from_aligned_fasta.pl EXET_seqs_01_10_20_to_28_02_21.fna $outbreak_file
  done

```

### Tidy up the files into directories

```{bash include=T, eval=T, echo=T, warning=T, message=T}
for i in nexus_files haplotype_pseudosequences haplotype_csv_files; do
  if [ -d $i ]; then
    echo $i directory exists
else
  echo creating directory $i
  mkdir $i
  fi
done
  
mv *.nex nexus_files/
mv *.outbreak.*.csv haplotype_csv_files/
mv *.pseudosequences*.fna haplotype_pseudosequences/
rm outbreak.*.tsv
rm patientstaff_data_for_haplotype_network.txt

```







